### PR TITLE
Fix missing args messages

### PR DESCRIPTION
**Pull request by martica**
Fleshes out error messages around missing and extra arguments.

## PR Checklist:

- [x] All new features have been tested
- [ ] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: tests/structures/test_class.py [hunk 1]
```diff
@@ -191,3 +191,57 @@ def second(self):
             print(obj.second())
             print('Done.')
             """)
+
+    def test_method_too_many_positional_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a):
+                print(a)
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 2]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
             """)
+
+    def test_method_too_many_positional_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a):
+                print(a)
+                
+            obj = MyClass()
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 3]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
+    def test_method_too_many_positional_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a):
+                print(a)
+                
+            obj = MyClass()
+                
+            obj.f(1, 2)
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 4]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
+            
+            print("Done.")
+        """)
+
+    def test_method_unknown_keyword_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a):
+                print(a)
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 5]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
+        """)
+
+    def test_method_unknown_keyword_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a):
+                print(a)
+                
+            obj = MyClass()
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 6]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
+    def test_method_unknown_keyword_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a):
+                print(a)
+                
+            obj = MyClass()
+                
+            obj.f(1, y=2)
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 7]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
+            print("Done.")
+        """)
+
+    @expectedFailure
+    def test_method_duplicated_keyword_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a, y=None):
+                print(a, y)
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 8]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
+    @expectedFailure
+    def test_method_duplicated_keyword_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a, y=None):
+                print(a, y)
+                
+            obj = MyClass()
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 9]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a, y=None):
+                print(a, y)
+                
+            obj = MyClass()
+                
+            x = {'y': 1}
+            obj.f(1, y=2, **x)
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 10]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
+            
+            print("Done.")
+        """)
+
+    def test_method_missing_positional_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a, b, c):
+                print(a, b, c)
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 11]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
+        """)
+
+    def test_method_missing_positional_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a, b, c):
+                print(a, b, c)
+                
+            obj = MyClass()
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_class.py [hunk 12]
```diff
@@ -191,3 +191,57 @@ def second(self):
...
+    def test_method_missing_positional_args(self):
+        self.assertCodeExecution("""
+            class MyClass:
+              def f(self, a, b, c):
+                print(a, b, c)
+                
+            obj = MyClass()
+                
+            obj.f(1)
+
```
Comment (brutusthebee) [inline tests/structures/test_class.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_function.py [hunk 13]
```diff
@@ -221,3 +221,63 @@ def myfunc(x, y, z):
             myfunc(**{'z': 4, 'x': 1, 'y': 2})
             print('Done.')
         """)
+
+    @unittest.expectedFailure
+    def test_missing_keyword_args(self):
+        self.assertCodeExecution("""
+            def myfunc(x, *, z):
+                print(x, z)
+
```
Comment (brutusthebee) [inline tests/structures/test_function.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_function.py [hunk 14]
```diff
@@ -221,3 +221,63 @@ def myfunc(x, y, z):
...
+            
+            myfunc(1)
+            print('Done.')
+        """)
+
+    def test_missing_positional_args(self):
+        self.assertCodeExecution("""
+            def myfunc(x, y, z):
+                print(x, y, z)
+
```
Comment (brutusthebee) [inline tests/structures/test_function.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_function.py [hunk 15]
```diff
@@ -221,3 +221,63 @@ def myfunc(x, y, z):
...
+            
+            myfunc()
+            print('Done.')
+        """)
+
+    def test_unexpected_keyword_arg(self):
+        self.assertCodeExecution("""
+            def myfunc(x, a=1):
+                print(x, a)
+
```
Comment (brutusthebee) [inline tests/structures/test_function.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_function.py [hunk 16]
```diff
@@ -221,3 +221,63 @@ def myfunc(x, y, z):
...
+            myfunc(1, b=2)
+            print('Done.')
+        """)
+
+    @unittest.expectedFailure
+    def test_repeated_keyword_arg(self):
+        self.assertCodeExecution("""
+            def myfunc(x, a=1):
+                print(x, a)
+
```
Comment (brutusthebee) [inline tests/structures/test_function.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_function.py [hunk 17]
```diff
@@ -221,3 +221,63 @@ def myfunc(x, y, z):
...
+            except Exception as e:
+                print(e)
+            print('Done.')
+        """)
+
+    def test_varargs(self):
+        self.assertCodeExecution("""
+            def myfunc(*a):
+                print(a)
+
```
Comment (brutusthebee) [inline tests/structures/test_function.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_function.py [hunk 18]
```diff
@@ -221,3 +221,63 @@ def myfunc(x, y, z):
...
+                
+            myfunc(3,4,5,6,7)
+            print('Done.')
+        """)
+
+    def test_too_many_positional_args(self):
+        self.assertCodeExecution("""
+            def myfunc(a):
+                print(a)
+
```
Comment (brutusthebee) [inline tests/structures/test_function.py]: At column 1: (W293) blank line contains whitespace

File: tests/structures/test_function.py [hunk 19]
```diff
@@ -221,3 +221,63 @@ def myfunc(x, y, z):
...
+    def test_too_many_positional_args(self):
+        self.assertCodeExecution("""
+            def myfunc(a):
+                print(a)
+                
+            myfunc(3,4,5,6,7)
+            print('Done.')
+        """)
+
```
Comment (brutusthebee) [inline tests/structures/test_function.py]: At column 1: (W391) blank line at end of file

File: tests/structures/test_function.py [hunk 20]
```diff
@@ -221,3 +221,62 @@ def myfunc(x, y, z):
...
+            myfunc()
+            print('Done.')
+        """)
+
+    def test_unexpected_keyword_arg(self):
+        self.assertCodeExecution("""
+            def myfunc(x, a=1):
+                print(x, a)
+
```
Comment (brutusthebee) [inline tests/structures/test_function.py]: At column 1: (W293) blank line contains whitespace

**Comment by martica** [comment]
This is ready to review. Supporting **kwargs and the like needs some of the new opcodes that are coming in #772, so it can wait.

**Pull request merged**